xla/service/gpu/gpu_compiler.cc

```diff
@@ -869,7 +869,12 @@ absl::Status RunCollectiveOptimizationPasses(
         CollectivePipeliner::PipeliningDirection::kBackward,
         /*should_process=*/HloPredicateIsOp<HloOpcode::kAllGather>,
         /*acceptable_formatting=*/HloPredicateTrue,
-        /*reuse_pipelined_op_buffer=*/HloPredicateFalse};
+        /*reuse_pipelined_op_buffer=*/HloPredicateFalse,
+        /*should_allow_loop_variant_parameter_in_chain=*/HloPredicateFalse,
+        /*should_allow_control_dependencies=*/false,
+        /*postprocess_backward_peeled_op=*/std::nullopt,
+        /*postprocess_backward_rotated_op=*/std::nullopt,
+        /*acceptable_loop_invariant_op_in_chain=*/true};
```

Comment:
Should this be an XLA flag?

Reply:
the intention is to let this be controlled by the pipeliner flag and support pipelining loop invariant by default for ag for gpu. I dont think having a separate flag for just this is necessary but let me know if you think otherwise.